source/common/stats/metric_impl.h

```diff
@@ -55,21 +55,15 @@ class MetricHelper {
  */
 template <class BaseClass> class MetricImpl : public BaseClass {
 public:
-  MetricImpl(absl::string_view name, absl::string_view tag_extracted_name,
-             const std::vector<Tag>& tags, SymbolTable& symbol_table)
-      : helper_(name, tag_extracted_name, tags, symbol_table) {}
-
-  // Alternate API to take the name as a StatName, which is needed at most call-sites.
-  // TODO(jmarantz): refactor impl to either be able to pass string_view at call-sites
-  // always, or to make it more efficient to populate a StatNameList with a mixture of
-  // StatName and string_view.
+  // TODO(jmarantz): Use StatName for tag_extracted_name.
   MetricImpl(StatName name, absl::string_view tag_extracted_name, const std::vector<Tag>& tags,
              SymbolTable& symbol_table)
-      : MetricImpl(symbol_table.toString(name), tag_extracted_name, tags, symbol_table) {}
```

Comment:
FYI this is where the bug was, because we lost the structure of the original StatName, and which components of it were dynamic, which matters for hash-tables.